smartlearner/optimizers/adagrad.py

```diff
@@ -0,0 +1,55 @@
+from collections import OrderedDict
+import theano.tensor as T
+
+from . import SGD
+from ..utils import sharedX
+
+
+class ADAGRAD(SGD):
+    """
+    Implements the ADAGRAD optimizer.
+
+    Reference
+    ---------
+    Duchi, J., Hazan, E., & Singer, Y. (2010).
+    Adaptive subgradient methods for online learning and stochastic optimization.
+    Journal of Machine Learning
+    """
```

Comment:
Do you really want to add an empty line before every closing """ ? It seems to me a bit overkill and it is clear enough without it (anyway when generating the documentation I pretty sure the `__docstring__` is trimmed).